RecoTracker/PixelSeeding/plugins/gpuPixelDoubletsAlgos.h

```diff
@@ -39,11 +39,33 @@ namespace gpuPixelDoublets {
     using H = HitsConstView<TrackerTraits>;
     using T = TrackerTraits;
 
-    const uint32_t maxNumberOfDoublets_;
-    const bool doClusterCut_;
-    const bool doZ0Cut_;
-    const bool doPtCut_;
-    const bool idealConditions_;  //this is actually not used by phase2
+    CellCutsT() = default;
+    CellCutsT(const bool doClusterCut,
+              const bool doZ0Cut,
+              const bool doPtCut,
+              const bool idealConditions,
+              const float z0Cut,
+              const float ptCut,
+              const std::vector<int>& phiCutsV)
+        : doClusterCut_(doClusterCut),
+          doZ0Cut_(doZ0Cut),
+          doPtCut_(doPtCut),
+          idealConditions_(idealConditions),
+          z0Cut_(z0Cut),
+          ptCut_(ptCut) {
+      assert(phiCutsV.size() == TrackerTraits::nPairs);
+      std::copy(phiCutsV.begin(), phiCutsV.end(), &phiCuts[0]);
+    }
+
+    bool doClusterCut_;
+    bool doZ0Cut_;
+    bool doPtCut_;
+    bool idealConditions_;  //this is actually not used by phase2
+
+    float z0Cut_;
+    float ptCut_;
```

Comment:
can these be `const` like before ?

Reply:
Let me check I don't remember if there was a reason for moving this to non-`const`

Reply:
Ok the problem is [here](https://github.com/cms-sw/cmssw/blob/4fac7bb211bfbbd03d35543221c44a7c3b1a8e1e/RecoTracker/PixelSeeding/plugins/CAHitNtupletGeneratorKernelsAlloc.cc#L52) I think. With `const` I can't assign, or I'm missing something?

Reply:
I see.

I think for the CPU case we could find a workaround, but for the GPU case it would probably be too complicated.

OK like this, then (maybe to be reviewed after the Alpaka migration and the use of buffers).